### DISCUSSION TITLE
Using branch.<name>.description as PR title

### DISCUSSION THREAD

**Discussion by DamienCassou** [magit/forge#513] [Ideas]
Would it be possible to use `branch.<name>.description` as title for a PR being created? This string could be automatically inserted into the forge-post buffer or it could be made accessible with a key binding.

**Comment by DamienCassou** [comment]
I volunteer to implement that if you give me guidance on the kind of UX you would like.

**Reply by tarsius** [reply]
I just added a hook that could be used for that, see #512.

The hook function probably shouldn't discard an existing template, but different users might feel differently about it (and some projects surely won't like it if contributors discard the template). Maybe that means it shouldn't be added to forge itself, or maybe it should.

(As mentioned in that pull-request, I currently don't respond to suggestions for the most part. That's why I haven't responded to your other suggestions.)

**Reply by DamienCassou** [reply]
Thank you very much. Unfortunately, I realize only now that descriptions are attached to local branches but PRs are made between remote branches only. I could try to guess the local branch name by extracting everything after the first `/`. What do you think?

**Reply by tarsius** [reply]
` (oref (forge-current-pullreq) head-ref)` seems like the way to go. It won't be correct in 100%, but if your feature branches have the same name locally and on your remote, then that should work.

**Reply by DamienCassou** [reply]
If the PR doesn't exist yet, `(forge-current-pullreq)` always seems to return `nil`.

**Reply by tarsius** [reply]
I've added `forge-create-pullreq-insert-branch-description`, intended to be added to `forge-edit-post-hook`.

**Reply by DamienCassou** [reply]
Thank you!
